### ISSUE TITLE
Allow modal dialogs to use no-cancel-on-outside-click=false

### ISSUE THREAD

**Issue by anthonytordillos** [PolymerElements/paper-dialog-behavior#23]
There are many cases where the best a11y experience is to have focus stay within the dialog, but where sighted users would still benefit from the ability to close the dialog by clicking outside of it. Can we modify the behavior of the modal attribute such that no-cancel-on-outside-click=false is respected?

**Comment by anthonytordillos** [comment]
Ping?

**Comment by samccone** [comment]
:+1: seems reasonable to me.

@ebidel thoughts around this one?

**Comment by ebidel** [comment]
This is @cdata 's element

**Comment by anthonytordillos** [comment]
Friendly ping.

**Comment by anthonytordillos** [comment]
Ping?

**Comment by valdrinkoshi** [comment]
Hi @anthonytordillos, `modal` sets for you `with-backdrop` and `no-cancel-on-outside-click`, and also blocks the focus change. 
If you'd like to have a different behavior, you can avoid setting `modal` and instead just the booleans you need, in this case only `with-backdrop`:
```html
<paper-dialog with-backdrop>
  Hello
</paper-dialog>
```
Also, remember that you cannot set properties to false from markup, since the presence of the attribute, with or without a value, equates to true. This is the standard behavior for attributes in the web platform.
https://www.polymer-project.org/1.0/docs/devguide/properties.html#configuring-boolean-properties

**Comment by anthonytordillos** [comment]
But I don't want just `with-backdrop`, I also want `modal`, I just *don't* want `no-cancel-on-outside-click`.

**Comment by valdrinkoshi** [comment]
I'm not sure I get the requirement. To recap:
- `with-backdrop` takes care of displaying the backdrop
- `no-cancel-on-outside-click` prevents an overlay to be closed when we click outside its bounds

`modal` sets both of the above in order to see the backdrop and block clicks outside its bounds. In addition, it will restore the focus to the focused element when you click outside the overlay bounds.

In other words: if you want `modal`, you need also `no-cancel-on-outside-click`.
If you want to enable the closing when user clicks outside the overlay bounds + the backdrop, all you have to do is set the `with-backdrop` property and nothing else :+1:

**Comment by anthonytordillos** [comment]
Specifically what we'd like is:

* `modal`, so that keyboard focus is constrained to the dialog (i.e. the user can't tab outside of the dialog)
* the ability to close the dialog by clicking outside of it

This desired behavior is driven by two groups:

- UX, which would like an easy and intuitive way for the majority of users to close dialogs.
- a11y, which would like focus to be constrained to the dialog while the dialog is open.

**Comment by valdrinkoshi** [comment]
This is a very valid point. I've opened an issue on `iron-overlay-behavior` since it should be fixed there https://github.com/PolymerElements/iron-overlay-behavior/issues/88
Also `paper-dialog-behavior` should be updated since it won't need to handle the focus trapping anymore
